projects/scion/workbench/src/lib/routing/workbench-url-observer.service.ts

```diff
@@ -208,14 +207,26 @@ export class WorkbenchUrlObserver {
   }
 
   /**
-   * Sets the workbench layout to {@link WorkbenchLayoutService}, but only if navigating through the workbench router.
-   *
-   * Note: Apply the layout after navigation ended to not run into the following error: 'Cannot activate an already activated outlet'.
+   * Updates registries and workbench handles to reflect the new layout.
    */
-  private applyWorkbenchLayout(): void {
-    const layout = this._workbenchRouter.getCurrentNavigationContext().layout;
+  private applyWorkbenchLayoutChanges(): void {
+    const {layout} = this._workbenchRouter.getCurrentNavigationContext();
+    if (layout !== this._workbenchLayoutService.layout) { // Layout instance does not change if navigating through the Angular router.
+      this.updateViewRegistry();
+      this.updatePartRegistry();
+
+      this._viewRegistry.views.forEach(view => view.onLayoutChange(layout));
```

Comment:
This could done in the `updateViewRegistry()`. Logically it belongs there, right?

Reply:
This is by intention. We first need to register the handles in the registries for the views/parts to access the associated model objects in `onLayoutChange'. For example, the view accesses its associated part. Therefore, the part must be registered before `onLayoutChange` of the view is invoked.